modules/app/components/ClientRenderOnly.tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react';
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const ClientRenderOnly: React.FC<Props> = ({ children, ...delegated }) => {
+  const [hasMounted, setHasMounted] = useState(false);
+
+  useEffect(() => {
+    setHasMounted(true);
+  }, []);
```

Comment:
i think here you can add a check "typeof window !== undefined" to make sure it's only client side

Reply:
Wrapping `setHasMounted`?

Reply:
> i think here you can add a check "typeof window !== undefined" to make sure it's only client side

I think it's safe without it.  Did a bit of digging to confirm that `useEffect` only runs for the first time after mounting so will never get run on the server.